mkosi/__init__.py

```diff
@@ -1442,6 +1442,38 @@ def want_signed_pcrs(config: Config) -> bool:
     )
 
 
+def add_ukify_secure_boot_arguments(
+    context: Context, out_cmd: list[PathString], out_options: list[PathString]
+) -> None:
+    assert context.config.secure_boot_key
+    assert context.config.secure_boot_certificate
+
+    if context.config.secure_boot_sign_tool != SecureBootSignTool.pesign:
+        out_cmd += [
+            "--signtool", "sbsign",
+            "--secureboot-private-key", context.config.secure_boot_key,
+            "--secureboot-certificate", context.config.secure_boot_certificate,
+        ]  # fmt: skip
+        out_options += [
+            "--ro-bind", context.config.secure_boot_certificate, context.config.secure_boot_certificate,
+        ]  # fmt: skip
+        if context.config.secure_boot_key_source.type == KeySourceType.engine:
+            out_cmd += ["--signing-engine", context.config.secure_boot_key_source.source]
+            out_options += ["--bind-try", "/run/pcscd", "/run/pcscd"]
+        if context.config.secure_boot_key.exists():
```

Comment:
Does it make sense to check the existence at this point? If the key is set and the key doesn't exist, then suppressing that seems like either a footgun or bound to blow up a bit later?

Reply:
I just copied that code out to a separate function, I don't know the reasoning behind this.

Reply:
The exists is to check if it's a pkcs11 uri or an actual file on disk